executor/builder.go

```diff
@@ -260,51 +278,79 @@ func (b *executorBuilder) buildIndexScan(v *plan.IndexScan) Executor {
 		} else {
 			ex = b.buildUnionScanExec(e)
 		}
+		ex = b.buildFilter(ex, v.AccessConditions)
```

Comment:
We don't need to buildFilter for `AccessConditions` if prefix index is not used.